### PR TITLE
Use labels to find intermediate images so they can be deleted

### PR DESCRIPTION
**Pull request by miklosduma**
Improvement recommendations for #106

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -12,6 +12,10 @@ FROM ${BUILDER_BASE_IMAGE} as builder
 # be released periodically.  Hence this builder image being built may
 # suffer a delay in security fixes and updates that is the sum of
 # those two: mitigate that by running `apt-get upgrade`.
```
Comment (lucafavatellawork) [inline docker/Dockerfile]: Please do not separate comment from context i.e. `apt-get upgrade`.

File: docker/Makefile [hunk 2]
```diff
@@ -34,8 +34,15 @@ DOCKER_CREATE_OPTS := -p8000:8000 -p9000:9000
...
 DOCKER_PS := docker ps -aqf ancestor=$(DOCKER_REPO):$(DOCKER_TAG)
-DOCKER_IMAGES := docker images -qf reference=$(DOCKER_REPO)
 
+# Use labels to find relevant final and intermediate images.
+# Intermediate image will have builder label (e.g. sse_builder=true),
+# and it will be "dangling", i.e. it has <none> tag.
+DOCKER_LABEL := sse_builder
+DOCKER_LABEL_BUILDER := true
+DOCKER_LABEL_RUNNER := false
+DOCKER_BUILDER_IMAGES := docker images -qf dangling=true -qf label=$(DOCKER_LABEL)=$(DOCKER_LABEL_BUILDER)
```
Comment (lucafavatellawork) [inline docker/Makefile]:
```suggestion
DOCKER_BUILDER_IMAGES := docker images -q -f dangling=true -f label=$(DOCKER_LABEL)=$(DOCKER_LABEL_BUILDER)
```

File: docker/Makefile [hunk 3]
```diff
@@ -34,8 +34,15 @@ DOCKER_CREATE_OPTS := -p8000:8000 -p9000:9000
...
+DOCKER_IMAGES := docker images -qf reference=$(DOCKER_REPO) -qf label=$(DOCKER_LABEL)=$(DOCKER_LABEL_RUNNER)
```
Comment (lucafavatellawork) [inline docker/Makefile]:
```suggestion
DOCKER_IMAGES := docker images -q -f reference=$(DOCKER_REPO) -f label=$(DOCKER_LABEL)=$(DOCKER_LABEL_RUNNER)
```

File: docker/Makefile [hunk 4]
```diff
@@ -47,6 +54,9 @@ docker:
 	--build-arg DEBIAN_VERSION="$(shell $(MAKE) -s -f $(ROOT_DIR)/deploy/Linux-x86_64/make_debian_vsn.mk debian-version)" \
 	--build-arg DEBIAN_DEB_REL_DIR=${DEBIAN_DEB_REL_DIR} \
 	--build-arg DEBIAN_DEB_NAME=${DEBIAN_DEB_NAME} \
+	--build-arg DOCKER_LABEL=${DOCKER_LABEL} \
```
Comment (lucafavatellawork) [inline docker/Makefile]: I expect target `dev_docker` needs update too?

**Comment by lucafavatellawork** [comment]
> Use labels to find intermediate images so they can be deleted

@miklosduma 

What do you mean by "deleted"?
And why do builder images need to be labelled?

Are you looking for some invocation like [`docker system prune`](https://docs.docker.com/engine/reference/commandline/system_prune/) or similar?

File: docker/Dockerfile (hunk 1 above)
Comment (miklosduma) [reply docker/Dockerfile]: Addressed by https://github.com/opensparkl/sse_core/pull/108/commits/8708dc2c2699c8930ca064625bd455e82aa8c76f

File: docker/Makefile (hunk 2 above)
Comment (miklosduma) [reply docker/Makefile]: Addressed by https://github.com/opensparkl/sse_core/pull/108/commits/a97112bd678792f2d1399fc284f5ad28eaf61f4b

File: docker/Makefile (hunk 4 above)
Comment (miklosduma) [reply docker/Makefile]: Addressed by https://github.com/opensparkl/sse_core/pull/108/commits/ebe090c2c8fb86f9cd9404f49a323297886f7ea9

**Comment by miklosduma** [comment]
@lucafavatellawork the multi-stage build creates multiple docker images. The intermediate images will not be tagged or referenced, so the `clean_docker` target does not delete them.

The aim of this modification is to get rid of the builder image too when running `clean_docker`.

**Comment by miklosduma** [comment]
Rebased on master

File: docker/Makefile [hunk 5]
```diff
@@ -124,13 +137,18 @@ help_docker:
 	@echo
 
 .PHONY: clean_docker
-clean_docker: clean_docker_containers clean_docker_images ;
+clean_docker: clean_docker_containers clean_docker_builder_images clean_docker_images ;
```
Comment (lucafavatellawork) [inline docker/Makefile]: Maybe `help_docker` shall be integrated.

Comment (lucafavatellawork) [reply docker/Makefile]: I am drafting something.

File: docker/Dockerfile [hunk 6]
```diff
@@ -42,7 +45,12 @@ RUN ( \
     make dist \
  && make -C deploy/Linux-x86_64 deb; \
     )
+
 FROM ${BASE_IMAGE}
+ARG DOCKER_LABEL
+ARG RUNNER_LABEL
+LABEL ${DOCKER_LABEL}=${RUNNER_LABEL}
```
Comment (lucafavatellawork) [inline docker/Dockerfile]: Adding this to the runner image seems unnecessary.

**Pull request merged**